File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/CollectionV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;

/**
 * Canonical collection/list container across all media types.
 */
public record CollectionV1(
        String collectionId,
        String ownerProfileId,
        String name,
        String description,
        String collectionType,
        String visibility,
        long createdAtEpochMs,
        long updatedAtEpochMs
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/CollectionV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract

data class CollectionV1(
    val collectionId: String,
    val ownerProfileId: String,
    val name: String,
    val description: String,
    val collectionType: String,
    val visibility: String,
    val createdAtEpochMs: Long,
    val updatedAtEpochMs: Long
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
